mouse.go

```diff
@@ -0,0 +1,118 @@
+package engi
+
+import (
+	"github.com/paked/engi/ecs"
+	"log"
+)
+
+// MouseComponent is the location for the MouseSystem to store its results; to be used / viewed by other Systems
+type MouseComponent struct {
+	// Clicked is true whenever the Mouse was space in this frame
+	Clicked bool
+	// Hovered is true whenever the Mouse is hovering the space in this frame
+	Hovered bool
+	// Dragged is true whenever the space was clicked, and then the mouse started moving (while holding)
+	Dragged bool
+	// Rightclicked is true whenever the space was rightclicked in this frame
+	Rightclicked bool
+	// Enter is true whenever the Mouse entered the space in that frame, but wasn't in that space on the previous frame
+	Enter bool
+	// Leave is true whenever the Mouse was in the space on the previous frame, but now isn't
+	Leave bool
+}
+
+func (*MouseComponent) Type() string {
```

Comment:
The `Type()` function? According to `golint`: yes, but tbh, if we were to actually comment *every* `Type()` function, it wouldn't make any sense..?